### PR TITLE
Player Piano Automatic Linking

### PR DESCRIPTION
**Pull request by Facsimil**
[FEATURE] [GAME OBJECTS] [QOL]

## About the PR 

Adds a new multitool function (based on #7167) that allows players to link a set of player pianos together.

## Why's this needed? 

It gets tedious linking pianos the larger the set is.

## Changelog 

```changelog
(u)Facsimil
(+)Player pianos now can be linked together automatically.
```

### CODE REVIEW

**Comment by Facsimil** [comment]
This is the first time I created a component. Sorry if I missed something.

**Comment by Facsimil** [comment]
¯ \ _ (ツ) _ / ¯

File: code/obj/playable_piano.dm [hunk 1]
```diff
@@ -207,6 +212,15 @@ TYPEINFO(/obj/player_piano)
 					return 1
 		return 0
 
+	proc/is_pulser_auto_linking(var/mob/M)
+		if(hasvar(M, "l_hand") && ispulsingtool(M.l_hand) && SEND_SIGNAL(M.l_hand, COMSIG_IS_PLAYER_PIANO_AUTO_LINKER_ACTIVE)) return TRUE
+		if(hasvar(M, "r_hand") && ispulsingtool(M.r_hand) && SEND_SIGNAL(M.r_hand, COMSIG_IS_PLAYER_PIANO_AUTO_LINKER_ACTIVE)) return TRUE
+		if(hasvar(M, "module_states"))
```
Comment (ZeWaka) [inline code/obj/playable_piano.dm:218]: no, typecheck

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/playable_piano.dm (hunk 1 above)
Comment (Facsimil) [reply code/obj/playable_piano.dm]: I added typechecks, though I am unsure if I understood correctly.

Comment (ZeWaka) [reply code/obj/playable_piano.dm]: no, as in don't use `hasvar`

File: code/obj/playable_piano.dm [hunk 2]
```diff
@@ -207,6 +212,15 @@ TYPEINFO(/obj/player_piano)
 					return 1
 		return 0
 
+	proc/is_pulser_auto_linking(var/mob/M)
+		if(ispulsingtool(M.l_hand) && SEND_SIGNAL(M.l_hand, COMSIG_IS_PLAYER_PIANO_AUTO_LINKER_ACTIVE)) return TRUE
+		if(ispulsingtool(M.r_hand) && SEND_SIGNAL(M.r_hand, COMSIG_IS_PLAYER_PIANO_AUTO_LINKER_ACTIVE)) return TRUE
+		if(istype(M, /mob/living/silicon))
+			for(var/atom/A in M:module_states)
```
Comment (ZeWaka) [inline code/obj/playable_piano.dm:219]: cast the mob as a silicon, `:` usage is forbidden

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/playable_piano.dm (hunk 2 above)
Comment (Facsimil) [reply code/obj/playable_piano.dm]: Removed semicolon operator. `module_states` is not a part of `/mob/living/silicon`, so checks are made separately for `/mob/living/silicon/robot` and `/mob/living/silicon/hivebot`.

**Comment by Facsimil** [comment]
woo 🎉

**Pull request merged**